driver-core/src/main/com/mongodb/client/model/expressions/ArrayExpression.java

```diff
@@ -51,28 +50,48 @@
      */
     <R extends Expression> ArrayExpression<R> map(Function<? super T, ? extends R> in);
 
+    IntegerExpression size();
+
+    BooleanExpression any(Function<T, BooleanExpression> map);
+
+    BooleanExpression all(Function<T, BooleanExpression> map);
+
+    NumberExpression sum(Function<T, NumberExpression> map);
+
+    NumberExpression multiply(Function<T, NumberExpression> map);
```

Comment:
Using `Function<T, NumberExpression> map` as a parameter of `multiply` (and other similar methods) is a clever way of working around the fact that in Java, unlike Rust (and, I am sure, other languages that unlike Java implement generics by duplicating code), we can't express that a method should be associated with a type only if the parametrization of the corresponding class meets certain criteria (in this particular case, the criterium is that only type argument is `NumberExpression`).

An alternative approach would have been to make `multiply` a `static` method that accepts `ArrayExpression<? extends NumberExpression>`. This way we would have directly associate the method with a type based on the criteria we need, at the cost of making the method `static`.

Reply:
But then multiply would not be chained, yes? I think even if it fits into the apply method, this would be odd/inconvenient.